### PR TITLE
feat: Create reactive composers for common presenter tasks

### LINKED ISSUE #303: Implement common rx composers
- Create a util functions to wrap observables
- Instead of manually loading and hiding, showing errors, use wrapper functions
- Implement interfaces to do this task

I'm working on it

### PR DESCRIPTION
**Pull request by iamareebjamal**
Fixes #303

- Create base presenter interfaces and implementations
- Create reactive composers
- Replace common logic throughout presenters

To be merged after #308

### CODE REVIEW

**Comment by iamareebjamal** [comment]
@mejariamol Review last commit

**Comment by iamareebjamal** [comment]
@arp95 @Shailesh351 @shreyasnbhat @rohanagarwal94 Please review last commit

File: app/src/main/java/org/fossasia/openevent/app/common/BaseDetailPresenter.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package org.fossasia.openevent.app.common;
+
+import android.support.annotation.CallSuper;
+
+import org.fossasia.openevent.app.common.contract.presenter.IDetailPresenter;
+
+public  abstract class BaseDetailPresenter<K, V> extends BasePresenter<V> implements IDetailPresenter<K, V> {
+
+    private K id;
```
Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/common/BaseDetailPresenter.java]: naming key `id` will make it's usage unclear. just pass multi parameters in `attach` and let the implementation save the key values accordingly. It shall reduce two base presenters into one.

**Comment by mariobehling** [comment]
Please resolve conflicts.

**Comment by iamareebjamal** [comment]
@arp95 @Shailesh351 @shreyasnbhat @rohanagarwal94 Please review

**Comment by mariobehling** [comment]
@arp95 @Shailesh351 @shreyasnbhat @rohanagarwal94 Please help with timely reviews and support your co-developers.

**Pull request merged**